### PR TITLE
PLENTY-261_use_new_rest_route

### PR DESCRIPTION
**Pull request by TheKeymaster**

### CODE REVIEW

File: src/Product.php [hunk 1]
```diff
@@ -421,20 +421,20 @@ public function processVariation(array $variation)
             return false;
         }
 
-        if ($variation['isMain'] || $this->getField('sort') === '') {
-            $this->setField('sort', $this->getFromArray($variation, 'position'));
+        if ($this->getField('sort') === '') {
+            $this->setField('sort', $this->getFromArray($variation['base'], 'position'));
```
Comment (dblaichinger) [inline src/Product.php:425]: Is `base` now the same as `isMain`?

File: src/Product.php [hunk 2]
```diff
@@ -519,13 +519,17 @@ public function processVariationsProperties(array $data)
             return $this;
         }
 
+        // TODO: Add this once we know how. Data like isSearchable, propertyGroupId, backendName are missing.
+        return $this;
```
Comment (dblaichinger) [inline src/Product.php:523]: What about the rest of the code below?

File: src/Product.php [hunk 3]
```diff
@@ -789,19 +793,15 @@ protected function getPropertyGroupForPropertyName($propertyGroupId)
      */
     protected function shouldProcessVariation(array $variation)
     {
-        if ($variation['isActive'] === false) {
+        if (isset($variation['base']['automaticListVisibility']) && $variation['base']['automaticListVisibility'] < 1) {
```
Comment (dblaichinger) [inline src/Product.php:796]: No `isActive` anymore?

File: src/Product.php [hunk 4]
```diff
@@ -1071,10 +1073,6 @@ protected function processTags(array $data)
         }
 
         foreach ($data as $tag) {
-            if ($tag['tagType'] !== 'variation') {
```
Comment (dblaichinger) [inline src/Product.php:1074]: This is now supported?

**Review by dblaichinger — CHANGES_REQUESTED** [review]
4 comments, build fixed in other story.

File: src/Product.php (hunk 2 above)
Comment (TheKeymaster) [reply src/Product.php:523]: Is fixed in 263.

File: src/Product.php (hunk 3 above)
Comment (TheKeymaster) [reply src/Product.php:796]: Is fixed in 263.

File: src/Product.php (hunk 4 above)
Comment (TheKeymaster) [reply src/Product.php:1074]: That is also fixed in 263 xD

File: src/Product.php (hunk 1 above)
Comment (TheKeymaster) [reply src/Product.php:425]: No base contains all data of the base product. But it is also merged with some information about the current variation. `$variation['base']['isMain']` is the same as before. This is also how it was implemented in 263.

File: src/Product.php (hunk 4 above)
Comment (dblaichinger) [reply src/Product.php:1074]: Still the same change after merge, how was it fixed in 263?

**Review by dblaichinger — CHANGES_REQUESTED** [review]

File: src/Product.php (hunk 4 above)
Comment (TheKeymaster) [reply src/Product.php:1074]: With the new route, the assigned tags that are assigned to variants are automatically of type variation. Therefore they've dropped this field entirely. See this response for example:

image

**Review by dblaichinger — APPROVED** [review]

**Pull request merged**